runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -63,15 +70,36 @@ private SdkComponents() {
    * unique ID for the {@link AppliedPTransform}. Multiple registrations of the same
    * {@link AppliedPTransform} will return the same unique ID.
    */
-  String registerPTransform(AppliedPTransform<?, ?, ?> pTransform) {
+  public String registerPTransform(
+      AppliedPTransform<?, ?, ?> pTransform, List<AppliedPTransform<?, ?, ?>> children)
+      throws IOException {
+    String name = registerPTransform(pTransform);
+    // If this transform is present in the components, nothing to do. return the existing name.
+    // Otherwise the transform must be translated and added to the components.
+    if (componentsBuilder.getTransformsOrDefault(name, null) != null) {
+      return name;
+    }
+    checkNotNull(children, "child nodes may not be null");
+    componentsBuilder.putTransforms(name, PTransforms.toProto(pTransform, children, this));
+    return name;
+  }
+
+  /**
+   * Gets the ID for the provided {@link AppliedPTransform}. The provided {@link AppliedPTransform}
+   * will not be added to the components produced by this {@link SdkComponents} until it is
+   * translated via {@link #registerPTransform(AppliedPTransform, List)}.
+   */
+  public String registerPTransform(AppliedPTransform<?, ?, ?> pTransform) {
```

Comment:
private?

Seems like nobody should call this method as its usage is an internal detail of this class